src/assets.zig

```diff
@@ -294,29 +302,66 @@ fn registerRecipesFromZon(zon: ZonElement) void {
 
 pub const Palette = struct { // MARK: Palette
 	palette: main.List([]const u8),
+
 	pub fn init(allocator: NeverFailingAllocator, zon: ZonElement, firstElement: ?[]const u8) !*Palette {
+		const self = switch(zon) {
+			.object => try loadFromZonLegacy(allocator, zon),
+			.array, .null => try loadFromZon(allocator, zon),
+			else => return error.InvalidPaletteFormat,
+		};
+
+		if(firstElement) |elem| {
+			if(self.palette.items.len == 0) {
+				self.palette.append(allocator.dupe(u8, elem));
+			}
+			std.debug.assert(std.mem.eql(u8, self.palette.items[0], elem));
+		}
+		return self;
+	}
+	fn loadFromZon(allocator: NeverFailingAllocator, zon: ZonElement) !*Palette {
+		const items = zon.toSlice();
+
 		const self = allocator.create(Palette);
 		self.* = Palette{
-			.palette = .init(allocator),
+			.palette = .initCapacity(allocator, @max(items.len, 128)),
```

Comment:
I think you can just use items.len here, the max seems to just adds additional complexity without even giving a clear performance benefit.
I do agree though that it makes sense to use initCapacity when the length is known.